hbase_storage/src/main/java/com/splicemachine/storage/HScan.java

```diff
@@ -240,4 +249,11 @@ public Scan unwrapDelegate(){
     public void setSmall(boolean small) {
         scan.setSmall(small);
     }
+
+    @Override
+    public void attachKeyPrefixFilter(DataValueDescriptor firstColumn, String tableVersion) throws IOException {
+        Filter secondaryFilter = scan.getFilter();
```

Comment:
Is it possible that we get e.g. a `FilterList` here instead of `MultiRowRangeFilter`? Could you please add some checks for this?

Reply:
This is not supposed to happen.  I've added the following here:

>         if (!(secondaryFilter instanceof MultiRowRangeFilter))
>             throw new IOException(StandardException.newException(LANG_INTERNAL_ERROR,
>                     "Expected KeyPrefixFilter's secondary filter to be a MultiRowRangeFilter."));